File: backend/parking-building-support-api/src/main/java/com/parkingbuilding/support/sharedreadmodel/entity/ParkingSessionImage.java
package com.parkingbuilding.support.sharedreadmodel.entity;

import java.time.OffsetDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "parking_session_images")
public class ParkingSessionImage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "session_id", nullable = false)
    private Long sessionId;

    @Column(name = "image_type", nullable = false)
    private String imageType; // ENTRY_PLATE, ENTRY_VEHICLE, EXIT_PLATE, EXIT_VEHICLE

    @Column(name = "image_url", nullable = false)
    private String imageUrl;

    @Column(name = "thumbnail_url")
    private String thumbnailUrl;

    @Column(name = "detected_plate_number")
    private String detectedPlateNumber;

    @Column(name = "created_at")
    private OffsetDateTime createdAt;

    @Column(name = "updated_at")
    private OffsetDateTime updatedAt;
}

File: backend/ParkingBuilding.CoreApi/Domain/Entities/ParkingSessionImage.cs
using System;

namespace ParkingBuilding.CoreApi.Domain.Entities
{
    public class ParkingSessionImage
    {
        public long Id { get; set; }
        public long SessionId { get; set; }
        public virtual ParkingSession Session { get; set; } = null!;

        public string ImageType { get; set; } = null!; // 'ENTRY_PLATE', 'ENTRY_VEHICLE', 'EXIT_PLATE', 'EXIT_VEHICLE'
        public string ImageUrl { get; set; } = null!;
        public string? ThumbnailUrl { get; set; }
        public string? DetectedPlateNumber { get; set; }
        public string? DetectedNormalizedPlateNumber { get; set; }
        public decimal? Confidence { get; set; }
        public bool IsPrimary { get; set; }

        public DateTimeOffset CapturedAt { get; set; } = DateTimeOffset.UtcNow;
        public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
        public DateTimeOffset UpdatedAt { get; set; } = DateTimeOffset.UtcNow;
    }
}
